arches/app/models/migrations/9648_add_empty_key_value_pairs_to_tiles.py

```diff
@@ -0,0 +1,20 @@
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("models", "9477_fix_for_spatial_view_dbf_function_edtf_displaying_null"),
+    ]
+
+    def forwards_func(apps, schema_editor):
+        TileModel = apps.get_model("models", "TileModel")
+
+        for tile in TileModel.objects.all():
+            tile.save()  # should trigger new code in `TileModel.save` method that adds empty key/value pairs to tile data
+
+    def reverse_func(apps, schema_editor):
+        pass
+
+    operations = [
+        migrations.RunPython(forwards_func, reverse_func),
+    ]
```

Comment:
It's a bit surprising but the updated save method isn't used here, so that code needs to be repeated here. These seem to work forward and backward.

```
    def forwards_func(apps, schema_editor):
        TileModel = apps.get_model("models", "TileModel")
        Node = apps.get_model("models", "Node")

        for tile in TileModel.objects.filter(data={}, provisionaledits__isnull=False):
            for node in Node.objects.filter(nodegroup_id=tile.nodegroup_id):
                if not str(node.pk) in tile.data:
                    tile.data[str(node.pk)] = None
            tile.save() 

    def reverse_func(apps, schema_editor):
        TileModel = apps.get_model("models", "TileModel")

        for tile in TileModel.objects.filter(provisionaledits__isnull=False):
            if bool(tile.provisionaledits and not any(tile.data.values())):
                tile.data = {}
                tile.save() 
```

Reply:
Ah -- that _is_ strange and I wonder why it worked on my local but not yours. I'm happy to instead of relying on the `Tile.save()` method to duplicate the logic here.  But I want to make sure we're thinking the same thing about the logic. 

There's a reason I didn't write a `reverse_func`. The `forwards_func` is straightforward but without collecting some metadata there's no way to reliably reverse the changes made to the database -- I assume there can be cases of blank k/v pairs in tile data before we apply these changes. It could be that it's safe to delete those k/v pairs on reverse but I wanted to err on the side of caution.

The functions listed above work well, but don't match the logic in the `TileModel` -- that is that the logic added to `TileModel` creates blank k/v pairs regardless of provisional edits.